### PR TITLE
Copy buffers less and access buffers directly.

### PR DESCRIPTION
**Pull request by manthey**
For the common use case of a source using a single array of data, take steps to avoid copying that data.  Prior to this change, when a source was set, a data buffer with a Float32Array was allocated for it.  This data buffer would then be inserted into the source, which internally would copy the Float32Array to a javascript array using a for loop.  Later, when the vertex buffer objects were created, the javascript array would be copied to a Float32Array.

Instead, if the source is not mutated, use the original Float32Array data buffer for the entire process.

This speeds up creation of data points in a test case by a factor of 3 (from an average time of 3.3 seconds down to 1.1 seconds for 300,000 points).

Additionally, the mapper object can return the internal Float32Array used for any source with a new getSourceBuffer method.  If this array is altered and dynamicDraw is turned on, then calling updateSourceBuffer without a data array will just reuse the source's internal Float32Array.  In this way, a calling program dos not need to create such an array itself.

### CODE REVIEW

**Comment by manthey** [comment]
@aashish24 I've included the allocate-less-memory code in this branch.

File: src/geomData.js [hunk 1]
```diff
@@ -604,11 +619,30 @@ vgl.sourceData = function(arg) {
     var i;
 
     //m_data = m_data.concat(data); //no, slow on Safari
+    /* If we will are given a Float32Array and don't have any other data, use
+     * it directly */
+    if (!m_data.length && data.length &&
+        Object.prototype.toString.call(data) === "[object Float32Array]") {
```
Comment (aashish24) [inline src/geomData.js]: curious, can we not use typeof here?

File: src/geomData.js [hunk 2]
```diff
@@ -604,11 +619,30 @@ vgl.sourceData = function(arg) {
...
+      m_data = data;
+      return;
+    }
+    /* If our internal array is immutable and we will need to change it, create
```
Comment (aashish24) [inline src/geomData.js]: Can we change commenting style to follow // (basically consistent with rest of the code)

File: src/geomData.js (hunk 1 above)
Comment (manthey) [inline src/geomData.js]: No.  `typeof` of a `Float32Array` returns `object`.  There might be a better way than what I used, but I haven't found it.

Comment (aashish24) [inline src/geomData.js]: Yes, I remember that. How about if (array instanceof Float32Array) ??

Comment (manthey) [inline src/geomData.js]: Yes, that is better.  I'll change to that.

Comment (aashish24) [inline src/geomData.js]: Thanks.

File: src/geomData.js (hunk 2 above)
Comment (manthey) [inline src/geomData.js]: I thought the common convention was to use `//` for single line comments and `/*` `*/` for multiple line comments.  The code uses `/*` `*/` for method documentation.

Comment (manthey) [inline src/geomData.js]: I'll change it, though.

Comment (aashish24) [inline src/geomData.js]: I am sorry, yes I guess we never changed completely to // because of jsdoc issues. Don't worry about those then.

Comment (aashish24) [inline src/geomData.js]: Please don't change -:)

Comment (jbeezley) [inline src/geomData.js]: For the record, jsdoc **only** parses comments starting with `/**`.

**Comment by aashish24** [comment]
LGTM thanks!

**Pull request merged**